depthai_bridge/src/TrackedFeaturesConverter.cpp

```diff
@@ -33,16 +33,14 @@ void TrackedFeaturesConverter::toRosMsg(std::shared_ptr<dai::TrackedFeatures> in
     msg.header.frame_id = _frameName;
     msg.features.resize(inFeatures->trackedFeatures.size());
 
-    for(const auto& feature : inFeatures->trackedFeatures) {
-        depthai_ros_msgs::msg::TrackedFeature ft;
-        ft.header = msg.header;
-        ft.position.x = feature.position.x;
-        ft.position.y = feature.position.y;
-        ft.age = feature.age;
-        ft.id = feature.id;
-        ft.harris_score = feature.harrisScore;
-        ft.tracking_error = feature.trackingError;
-        msg.features.emplace_back(ft);
```

Comment:
What goes wrong is the combination of `msg.features.resize(inFeatures->trackedFeatures.size())` and `msg.features.emplace_back(ft)`. [`msg.features.resize(.)`](https://cplusplus.com/reference/vector/vector/resize/) expands the vector with value-initialized elements (hence all the zero features). Later on, [`msg.features.emplace_back(.)`](https://cplusplus.com/reference/vector/vector/emplace_back/) appends the actual features.

So replacing `msg.features.resize(inFeatures->trackedFeatures.size())` with [`msg.features.reserve(inFeatures->trackedFeatures.size())`](https://cplusplus.com/reference/vector/vector/reserve/) might solve the problem too.

Reply:
I agree with @wouterio, although [vec[i] might be faster than emplace_back in some cases](https://johnnysswlab.com/what-is-faster-vec-emplace_backx-or-vecx/#:~:text=For%20the%20emplace_back()%20we,the%20vector%20before%20using%20it.), I would leave it as is and only change the `resize` part

Reply:
He @Serafadam, thanks for addressing this!

Reply:
I also expect `emplace_back` to be the right choice here, because the link you mentioned concludes with this:

> In principle, if the type stored in the vector is a simple type (not a class or a struct), the loop is often vectorizable and one could expect to see similar numbers as we have seen here. With larger classes, it is the opposite: vectorization doesn’t pay off because of the low memory efficiency, so the operator[] version will in principle slower.

Since the features we're inserting are class/struct types, so we can expect `emplace_back` to be faster. But of course:

> as always with software performance, the runtime numbers are the ones that give the final verdict.

And that's up to you...